### PR TITLE
DAOS-9651 vos: Add VOS aggregation metrics

### PR DESCRIPTION
**Pull request by NiuYawei**
Added per pool VOS aggregation metrics.

Signed-off-by: Niu Yawei <yawei.niu@intel.com>

### CODE REVIEW

**Review by jolivier23 — APPROVED** [review]

File: src/vos/vos_internal.h [hunk 1]
```diff
@@ -1090,9 +1111,9 @@ gc_reserve_space(daos_size_t *rsrvd);
  * \param range_discard[IN]	Discard only uncommitted ilog entries (for reintegration)
  *
  * \return		Zero on Success
- *			1 if a reprobe is needed (entry is removed or not
- *			visible)
- *			negative value otherwise
+ *			Positive value if a reprobe is needed
+ *			(1: entry is removed; 2: entry is invisible)
```
Comment (Nasf-Fan) [inline src/vos/vos_internal.h:1115]: Better to define some MACRO for the return values.

**Review by Nasf-Fan — APPROVED** [review]

**Comment by johannlombardi** [comment]
IIRC, some other files must be edited when adding new metrics. @kjacque any thoughts?

**Review by kjacque — CHANGES_REQUESTED** [review]
It is also necessary to update `src/tests/ftest/util/telemetry_utils.py` with the new metrics. It would be wise to run with test-tag "telemetry" if you haven't already.

It looks like a number of the telemetry tests are already run with PRs. Unless they are being skipped, it's surprising to me that you wouldn't encounter failures with these changes. Have you verified your new metrics via both dmg telemetry metric list and daos_metrics?

**Comment by NiuYawei** [comment]
> It is also necessary to update `src/tests/ftest/util/telemetry_utils.py` with the new metrics. It would be wise to run with test-tag "telemetry" if you haven't already.
> 
> It looks like a number of the telemetry tests are already run with PRs. Unless they are being skipped, it's surprising to me that you wouldn't encounter failures with these changes. Have you verified your new metrics via both dmg telemetry metric list and daos_metrics?

I think the 'src/tests/ftest/util/telemetry_utils.py' only needs be modified when you adding a new system-wide metric (like when I was adding metrics for scheduler, I added scheduler metrics in this file), for per-pool metrics, I think this file doesn't need be touched. (Of course, I think the test could be improved to verify per-pool metrics as well, but that should be done in a separate patch)

Sure, I verified both 'daos_metrics' and 'dmg telemetry metrics list/query'.

**Comment by kjacque** [comment]
> I think the 'src/tests/ftest/util/telemetry_utils.py' only needs be modified when you adding a new system-wide metric (like when I was adding metrics for scheduler, I added scheduler metrics in this file), for per-pool metrics, I think this file doesn't need be touched. (Of course, I think the test could be improved to verify per-pool metrics as well, but that should be done in a separate patch)

You're right, seems like the metrics list test isn't getting the per-pool metrics. My bad. I think this is fine, then, as long as the metrics themselves look good coming out of dmg.

**Review by kjacque — APPROVED** [review]

**Pull request merged**